config/config.go

```diff
@@ -770,9 +770,15 @@ func initConfig(p koanf.Provider) (*Config, error) {
 	}
 
 	// Load environment variables and merge into the loaded config.
-	if err := k.Load(env.Provider("", ".", func(s string) string {
+	if err := k.Load(env.ProviderWithValue("", ".", func(key string, value string) (string, interface{}) {
 		// `__` is used as a hierarchy delimiter.
-		return strings.ReplaceAll(strings.ToLower(s), "__", ".")
+		key = strings.ReplaceAll(strings.ToLower(key), "__", ".")
+
+		// If vlaue contains a comma, split it into a list.
+		if strings.Contains(value, ",") {
```

Comment:
It would be safer to explicitly check for the key and only split, if the key is the one that should result in `[]string`.

It is not just theoretically unsafe. The other configuration option that we pass via env, is the database connection string. This one can contain commas, but should remain a string, not `[]string`, e.g.:

```text
postgresql://host1:123,host2:456/somedb?target_session_attrs=any&application_name=myapp
```

Reply:
Updated